src/transformers/processing_utils.py

```diff
@@ -54,6 +68,220 @@
+    input_data_format: Optional[Union[str, ChannelDimension]]
+
+
+class AudioKwargs(TypedDict, total=False):
+    """
+    Keyword arguments for audio processing.
+
+    Attributes:
+        sampling_rate (`int`, *optional*):
+            The sampling rate at which the `raw_speech` input was sampled.
+        raw_speech (`np.ndarray`, `List[float]`, `List[np.ndarray]`, `List[List[float]]`):
+            The sequence or batch of sequences to be padded. Each sequence can be a numpy array, a list of float
+            values, a list of numpy arrays or a list of list of float values. Must be mono channel audio, not
+            stereo, i.e. single float per timestep.
+        padding (`bool`, `str` or [`~utils.PaddingStrategy`], *optional*):
+            Select a strategy to pad the returned sequences (according to the model's padding side and padding
+            index) among:
+
+            - `True` or `'longest'`: Pad to the longest sequence in the batch (or no padding if only a single
+                sequence if provided).
+            - `'max_length'`: Pad to a maximum length specified with the argument `max_length` or to the maximum
+                acceptable input length for the model if that argument is not provided.
+            - `False` or `'do_not_pad'` (default): No padding (i.e., can output a batch with sequences of different
+                lengths).
+        max_length (`int`, *optional*):
+            Maximum length of the returned list and optionally padding length (see above).
+        truncation (`bool`):
+            Activates truncation to cut input sequences longer than *max_length* to *max_length*.
+        pad_to_multiple_of (`int`, *optional*):
+            If set, will pad the sequence to a multiple of the provided value.
+        return_attention_mask (`bool`, *optional*, defaults to `False`):
+            Whether or not [`~ASTFeatureExtractor.__call__`] should return `attention_mask`.
+    """
+
+    sampling_rate: Optional[int]
+    raw_speech: Optional[Union["np.ndarray", List[float], List["np.ndarray"], List[List[float]]]]
+    padding: Optional[Union[bool, str, PaddingStrategy]]
+    max_length: Optional[int]
+    truncation: Optional[bool]
+    pad_to_multiple_of: Optional[int]
+    return_attention_mask: Optional[bool]
+
+
+class CommonKwargs(TypedDict, total=False):
+    return_tensors: Optional[Union[str, TensorType]]
+
+
+class ProcessingKwargs(TextKwargs, ImagesKwargs, VideosKwargs, AudioKwargs, CommonKwargs, total=False):
+    """
+    Base class for kwargs passing to processors.
+    A model should have its own `ModelProcessorKwargs` class that inherits from `ProcessingKwargs` to provide:
+        1) Additional, typed keys and that this model requires to process inputs.
+        2) Default values for existing keys.
+    New keys have to be defined as follows to ensure type hinting is done correctly.
+
+    ```python
+    images_kwargs: ImagesKwargs = {"new_image_kwarg": Optional[bool]}
```

Comment:
Should it be? 

```suggestion
    images_kwargs: ImagesKwargs = {
        **ImagesKwargs.__annotations__,
        "new_image_kwarg": Optional[bool],
    }
```

Reply:
Docstring is outdated, will correct it, the way to add it is actually to add a class inheriting from ImagesKwargs:

```python
class AlignImagesKwargs(ImagesKwargs, total=False):
    new_image_kwarg: Optional[bool]

class AlignProcessorKwargs(ProcessingKwargs, total=False):
    images_kwargs: AlignImagesKwargs
    _defaults = {
        "text_kwargs": {
            "padding": "max_length",
            "max_length": 64,
        },
    }

```

wdyt? this preserves previous type hints
![image](https://github.com/huggingface/transformers/assets/39954772/c68d54f6-237a-4ef2-aea5-0552aa8ee102)
as well as new ones
![image](https://github.com/huggingface/transformers/assets/39954772/a4efc496-7392-4fa4-a702-bf4a723f31a1)

Reply:
Looks good 👍